packages/libuild/src/plugins/style/cssPlugin.ts

```diff
@@ -29,31 +30,22 @@ export const cssPlugin = (): LibuildPlugin => {
       compiler.hooks.transform.tapPromise(pluginName, async (source): Promise<Source> => {
         if (isStyleExt(source.path)) {
           const { query } = resolvePathAndQuery(source.path);
-          if (query[cssVirtual]) {
-            return {
-              code: source.code,
-              loader: 'css',
-              path: source.path,
-            };
+          let { code, loader = 'css' } = source;
+          if (!query[cssVirtual]) {
+            ({ code, loader } = await transformStyle.apply(compiler, [source]));
           }
-          const { contents, modules } = await transformStyle.apply(compiler, [source]);
-          if (Object.values(modules).length) {
-            // add hash query for same path, let esbuild cache invalid
-            const code = `
-import "${source.path}?css_virtual&hash=${getHash(contents, 'utf-8')}";
-export default ${JSON.stringify(modules)}`;
-            moduleCache[source.path] = contents;
-            return {
-              ...source,
-              code,
-              loader: 'js',
-            };
+          if (compiler.config.style.inject && loader === 'css') {
+            const styleInjectPath = require.resolve('style-inject/dist/style-inject.es').replace(/[\\/]+/g, '/');
+            const cssVariableName = identifier('css', true);
+            code = `var ${cssVariableName} = ${JSON.stringify(
+              code
```

Comment:
Use `JSON.stringify` for `code`, I think this is what will happen:

input:
``` css
.box {
   color: blue;
}
```
output:

``` html
<style type="text/css">"\n.a {\n    color: blue;\n}\n"</style>
```